### ISSUE TITLE
Gradient Tracking and Broadcasting for Arithmetic Operations similar to torch.nn.parameter.Parameter for kornia.geometry.quaternion.Quaternion

### ISSUE THREAD

**Issue by bn-nb** [kornia/kornia#3101]
## Feature

`kornia.geometry.quaternion.Quaternion` supports multiplication with other `Quaternion`s only. Even if a compatible type is passed for multiplication, no conversion attempt to `torch.Tensor` is made. A solution is to convert it to tensor as a Scalar `Quaternion` with zero vector components, if it is broadcastable with the real component of Quaternion ( `self.real.shape == ( self.real + torch.tensor(right) ).shape`)

One major implication, is to reflect this behaviour in all basic arithmetic operations, and also to wrap operators to allow reverse operation. For example, if `left` is some `torch.Tensor` compatible data type, and `quat` is an instance of `Quaternion`, `left` is first converted to a Scalar  `Quaternion`, and its corresponding __mul__ is called.

One solution I can think of is for `Quaternion` to be written to inherit from `Parameter` instead of `Module`, since `Parameter` class initializes the `self.data` attribute by itself.

## Motivation

`torch.Parameter` allows such broadcasting arithmetic, even if `requires_grad=True`. Granted that in-place operations must not be allowed. My goal is to be able to use this instances of `Quaternion` to use as trainable parameters in quaternion-space neural networks, that will support both real and quaternion inputs.

Even basic arithmetic operations, such as multiplication do not support gradients, as they do not use instances of `torch.autograd.Function` on `self.data`

## Alternatives

If this feature does not seem relevant to this project, one solution may be to manually check broadcastability with `quat.data` and perform `quat = Quaternion(quat.data * 5)` but this will invalidate gradient computation. Another alternative is to always augment the input data to Quaternions in a way such that only Quaternion products are involved. This will still not track gradients.

## Code

```python
import torch   # 2.5.1+cpu
import kornia  # 0.7.4

test1 = torch.randn((2,2,4))

p = torch.nn.parameter.Parameter(test1)
q = kornia.geometry.quaternion.Quaternion(test1)

a = p * p * 5
print(a)
print(a.grad_fn)  # MulBackward0

b = q * q
print(b.data.grad_fn) # None

c = q * q * 5 # TypeError
```

**Comment by edgarriba** [comment]
Sounds good to me. What do you think @cjpurackal ?

**Comment by Meggison** [comment]
@edgarriba I can help take on this issue
